test/NuGet.Core.Tests/NuGet.Commands.Test/RestoreCommandTests.cs

```diff
@@ -1416,6 +1417,215 @@ public async Task RestoreCommand_CentralVersion_ErrorWhenCentralPackageVersionFi
+            var provider = new DependencyProvider();
+            // D is a transitive dependency for package A through package B -> C -> D
+            // D is defined as a Central Package Version
+            // In this context Package D with version centralPackageVersion will be added as inner node of Node A, next to B 
+
+            // Input 
+            // A -> B (version = 3.0.0) -> C (version = 3.0.0) -> D (version = 3.0.0)
+            // A ~> D (version = 2.0.0
+            //         the dependency is not direct,
+            //         it simulates the fact that there is a centrally defined "D" package
+            //         the information is added to the provider)
+
+            // The expected output graph
+            //    -> B (version = 3.0.0) -> C (version = 3.0.0)
+            // A
+            //    -> D (version = 2.0.0)
+            provider.Package("A", otherVersion)
+                    .DependsOn("B", otherVersion)
+                    .DependsOn(centralPackageName, centralPackageVersion, LibraryDependencyTarget.Package, versionCentrallyManaged: true);
+
+            provider.Package("B", otherVersion)
+                   .DependsOn("C", otherVersion);
+
+            provider.Package("C", otherVersion)
+                  .DependsOn(centralPackageName, otherVersion);
+
+            // Simulates the existence of a D centrally defined package that is not direct dependency
+            provider.Package("A", otherVersion)
+                     .DependsOn(centralPackageName, centralPackageVersion, LibraryDependencyTarget.Package, versionCentrallyManaged: true);
+
+            // Add central package to the source with multiple versions
+            provider.Package(centralPackageName, "1.0.0");
+            provider.Package(centralPackageName, centralPackageVersion);
+            provider.Package(centralPackageName, "3.0.0");
+
+            context.LocalLibraryProviders.Add(provider);
+            var walker = new RemoteDependencyWalker(context);
+
+            // Act
+            var rootNode = await DoWalkAsync(walker, "A", framework);
+            RestoreTargetGraph restoreTargetGraph = RestoreTargetGraph.Create(new List<GraphNode<RemoteResolveResult>>() { rootNode }, context, logger, framework);
+
+            await RestoreCommand.LogDowngradeWarningsOrErrorsAsync(new List<RestoreTargetGraph>() { restoreTargetGraph }, logger);
+
+            // Assert
+            Assert.Equal(1, logger.Errors);
+            Assert.Equal(1, logger.LogMessages.Count);
+            var logMessage = logger.LogMessages.First();
+            Assert.Equal(LogLevel.Error, logMessage.Level);
+            Assert.True(logMessage.Message.Contains("Detected package downgrade: D from 3.0.0 to centrally defined 2.0.0. "));
+            Assert.Equal(NuGetLogCode.NU1109, logMessage.Code);
+        }
+
+        [Fact]
+        public async Task RestoreCommand_DowngradeIsErrorWhen_DowngradedByCentralTransitiveDependency()
```

Comment:
Discussed during the meeting, we can add a bunch more tests scenarios.
We should probably create an issue for that.

Reply:
Added https://github.com/NuGet/Home/issues/9318